src/slider-new/SliderThumb.ts

```diff
@@ -0,0 +1,153 @@
+import * as React from "react";
+import { useLocale } from "@react-aria/i18n";
+import { BoxHTMLProps, BoxOptions, useBox } from "reakit";
+import { createComponent, createHook } from "reakit-system";
+import { focusWithoutScrolling, mergeProps } from "@react-aria/utils";
+
+import { clamp } from "../utils";
+import { SLIDER_THUMB_KEYS } from "./__keys";
+import { SliderStateReturn } from "./SliderState";
+import { useGlobalListeners, useMove } from "./helpers";
+
+export type SliderThumbOptions = BoxOptions &
+  Pick<
+    SliderStateReturn,
+    | "inputs"
+    | "orientation"
+    | "setThumbEditable"
+    | "isDisabled"
+    | "focusedThumb"
+    | "setThumbDragging"
+    | "trackRef"
+    | "getThumbValue"
+    | "setThumbValue"
+    | "setThumbPercent"
+    | "getThumbPercent"
+    | "step"
+  > & {
+    index: number;
+  };
+
+export type SliderThumbHTMLProps = BoxHTMLProps;
+
+export type SliderThumbProps = SliderThumbOptions & SliderThumbHTMLProps;
+
+export const useSliderThumb = createHook<
+  SliderThumbOptions,
+  SliderThumbHTMLProps
+>({
+  name: "SliderThumb",
+  compose: useBox,
+  keys: SLIDER_THUMB_KEYS,
+
+  useProps(options, htmlProps) {
+    const { index, inputs } = options;
+    const isVertical = options.orientation === "vertical";
+    const { direction } = useLocale();
+    const { addGlobalListener, removeGlobalListener } = useGlobalListeners();
+
+    // Immediately register editability with the state
+    options.setThumbEditable(index, !options.isDisabled);
```

Comment:
I think we should put this inside an useEffect to ensure it runs only once when the component mounts, not on each render.

```js
    React.useEffect(() => {
      // Immediately register editability with the state
      options.setThumbEditable(index, !options.isDisabled);

       // maybe put options.isDisabled in the deps
      // eslint-disable-next-line react-hooks/exhaustive-deps
    }, []);
```

Reply:
Yeah, we can do that 🤔 
I will check and apply the refactor.